### PR TITLE
feat: add MergeJoinRel

### PR DESCRIPTION
**Pull request by danepitkin**
Closes https://github.com/substrait-io/substrait-java/issues/198

### CODE REVIEW

**Comment by danepitkin** [comment]
The `MergeJoin` implementation is identical to the `HashJoin` implementation. Any changes unrelated to `MergeJoin` that you see are just code snippets being moved around (I grouped the three physical operator implementations together for better readability).

File: core/src/main/java/io/substrait/relation/ProtoRelConverter.java [hunk 1]
```diff
@@ -103,6 +102,12 @@ public Rel from(io.substrait.proto.Rel rel) {
       case HASH_JOIN -> {
         return newHashJoin(rel.getHashJoin());
       }
+      case NESTED_LOOP_JOIN -> {
```
Comment (EpsilonPrime) [inline core/src/main/java/io/substrait/relation/ProtoRelConverter.java:105]: Why is the order Hash, Nested, Merge over alphabetical sort?

Comment (danepitkin) [reply core/src/main/java/io/substrait/relation/ProtoRelConverter.java:105]: This is the order in the Substrait spec, but I agree alphabetical would be better. Will update!

Comment (EpsilonPrime) [reply core/src/main/java/io/substrait/relation/ProtoRelConverter.java:105]: And here.

**Review by EpsilonPrime — APPROVED** [review]

File: core/src/main/java/io/substrait/relation/ProtoRelConverter.java (hunk 1 above)
Comment (danepitkin) [reply core/src/main/java/io/substrait/relation/ProtoRelConverter.java:105]: oof, thank you

**Review by vbarua — APPROVED** [review]
Change looks good to me! Thanks for the cleanup.

**Pull request merged**